pkg/secretpath/path.go

```diff
@@ -0,0 +1,226 @@
+// Package secretpath implements utility functions for manipulating paths
+// compatible with SecretHub, e.g. namespaces, repositories, directories,
+// secrets and versions.
+package secretpath
+
+import (
+	"strconv"
+	"strings"
+
+	"bitbucket.org/zombiezen/cardcpx/natsort"
+)
+
+const (
+	elemSepByte    = '/'
+	elemSep        = string(elemSepByte)
+	versionSepByte = ':'
+	versionSep     = string(versionSepByte)
+	latestSuffix   = ":latest"
+)
+
+// Join joins any number of elements into a path, adding a separator
+// if necessary. Empty string elements are ignored.
+func Join(elements ...string) string {
+	result := ""
+	for _, e := range elements {
+		e = strings.Trim(e, elemSep)
```

Comment:
These checks and cleaning are redundant, as this will be cleaned by `Clean` as well.

```Go
result := ""
for _, e := range elements {
	result += elemSep + e
}

return Clean(result)
```
is sufficient.

---

If you do want to keep them in, they can be refactored like this to improve readability:

```Go
result := ""
for _, e := range elements {
	e = strings.Trim(e, elemSep)
	if len(e) == 0 {
		continue
	}
	if len(result) == 0 {
		result += e
	} else {
		result += elemSep + e
	}
}

return Clean(result)
```